cli/cli.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+
+"""
+CLI for interfacing with deployed analysis runner.
+See README.md for more information.
+"""
+import re
+import logging
+import subprocess
+
+import click
+import requests
+import google.auth
+import google.auth.transport.requests
+
+logging.basicConfig(level='INFO')
+
+BRANCH = 'main'
+DEFAULT_SERVER_LOOKUP = (
+    f'https://raw.githubusercontent.com/'
+    f'populationgenomics/analysis-runner/{BRANCH}/cli/servermap.json'
+)
+# Make this None if you want to support all organisations
+SUPPORTED_ORGANISATIONS = {'populationgenomics'}
+
+
+@click.command(
+    help='CLI for the analysis runner - a CPG service for running analysis from '
+    'some GitHub repository (at a specific commit). The parameters are used to form a '
+    'POST request, sent to a server based on the --dataset parameter you provide.'
+)
+@click.option(
+    '--dataset',
+    required=True,
+    help='The dataset name, which determines which '
+    'analysis-runner server to send the request to',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    required=True,
+    help='The output directory of the run, MUST start with gs://',
+)
+@click.option(
+    '--repository',
+    '--repo',
+    help='The URI of the repository to run, must be approved by the appropriate server.'
+    ' Default behaviour is to find the repository of the current working'
+    ' directory with `git remote get-url origin`',
+)
+@click.option(
+    '--commit',
```

Comment:
Might be worth mentioning that this also supports tags. However, `HEAD` is not allowed (and we should probably add more checks to e.g. disallow `HEAD^` and similar).